### ISSUE TITLE
Working?

### ISSUE THREAD

**Issue by tahirkhan0x** [and-89/CAN-BUS-emulator_arduino#1]
Is this working?

**Comment by and-89** [comment]
Yes, it works, but of course you need to define the messages which you wanna emulates.

**Comment by iamtahirkhan6** [comment]
Define as in? What do you mean?

Also should we remove the sensor wires?

**Comment by tahirkhan0x** [comment]
For SCR emulation, what messages do we have to emulate?

**Comment by and-89** [comment]
The best way is to read-out the messages direct from the can-bus, filter it, and put it into this code. If you want to find the message from SCR, you can perform two records, one with ACM connected and one without ACM then you will seen which messages "disappear"  from can bus.

**Comment by tahirkhan0x** [comment]
How do we successfully emulate the NOx+ AdBlue dosing? I mean the ECM coordinates how much to dose, right?

So, how do we bypass the Aftertreatment? By fully emulating the temperature, nox and pressure sensors or by manipulating them and showing to them that temp is less then 200deg celcius and it dosing cannot be performed?

Thank you! If you would like then i can pay you for the help!

**Comment by and-89** [comment]
Please send me more information on mail: and013412@gmail.com. I need to know which vehical you want to emulate.
